pkg/utils/grpc.go

```diff
@@ -35,13 +24,13 @@ func GetUser(userID *user.UserId, gwc gateway.GatewayAPIClient, machineAuthAPIKe
 	return getUserResponse.GetUser(), nil
 }
 
-// ImpersonateUser impersonates the given user
-func ImpersonateUser(usr *user.User, gwc gateway.GatewayAPIClient, machineAuthAPIKey string) (context.Context, error) {
-	ctx := revactx.ContextSetUser(context.Background(), usr)
+// ImpersonateServiceUser impersonates the given user
+func ImpersonateServiceUser(serviceUserID string, gwc gateway.GatewayAPIClient, serviceUserSecret string) (context.Context, error) {
```

Comment:
I wonder if `ImpersonateServiceUser` is really a good name. Isn't it more a `GetServiceUserContext()` function now?

Reply:
Thats what it is, yes :+1: